test/integration/elasticquota_controller_test.go

```diff
@@ -0,0 +1,355 @@
+	schedInformerFactory := schedformers.NewSharedInformerFactory(extClient, 0)
+	eqInformer := schedInformerFactory.Scheduling().V1alpha1().ElasticQuotas()
+
+	coreInformerFactory := informers.NewSharedInformerFactory(cs, 0)
+	podInformer := coreInformerFactory.Core().V1().Pods()
+	eqCtrl := controller.NewElasticQuotaController(cs, eqInformer, podInformer, extClient)
+	runtime.Must(schedv1alpha1.AddToScheme(scheme.Scheme))
+	if err = wait.Poll(100*time.Millisecond, 3*time.Second, func() (done bool, err error) {
+		groupList, _, err := cs.ServerGroupsAndResources()
+		if err != nil {
+			return false, nil
+		}
+		for _, group := range groupList {
+			if group.Name == scheduling.GroupName {
+				return true, nil
+			}
+		}
+		t.Log("waiting for crd api ready")
+		return false, nil
+	}); err != nil {
+		t.Fatalf("Waiting for crd read time out: %v", err)
+	}
+
+	// start controller
+	stopCh := server.SetupSignalHandler()
+	go eqCtrl.Run(1, ctx.Done())
+	schedInformerFactory.Start(stopCh)
+	coreInformerFactory.Start(stopCh)
+
+	testCtx.ClientSet = cs
+	testCtx = util.InitTestSchedulerWithOptions(
+		t,
+		testCtx,
+		true,
+		scheduler.WithFrameworkOutOfTreeRegistry(registry),
+	)
+	t.Log("init scheduler success")
+	defer testutil.CleanupTest(t, testCtx)
+
+	// Create a Node.
+	nodeName := "fake-node"
+	node := st.MakeNode().Name(nodeName).Label("node", nodeName).Obj()
+	node.Status.Allocatable = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(300, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewQuantity(300, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(3000, resource.DecimalSI),
+	}
+	node.Status.Capacity = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(300, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewQuantity(300, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(3000, resource.DecimalSI),
+	}
+	node, err = cs.CoreV1().Nodes().Create(ctx, node, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create Node %q: %v", nodeName, err)
+	}
+
+	for _, ns := range []string{"ns1", "ns2"} {
+		_, err := cs.CoreV1().Namespaces().Create(ctx, &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{Name: ns}}, metav1.CreateOptions{})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			t.Fatalf("Failed to integration test ns: %v", err)
+		}
+		autoCreate := false
+		t.Logf("namespaces %+v", ns)
+		_, err = cs.CoreV1().ServiceAccounts(ns).Create(ctx, &v1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{Name: "default", Namespace: ns}, AutomountServiceAccountToken: &autoCreate}, metav1.CreateOptions{})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			t.Fatalf("Failed to create ns default: %v", err)
+		}
+	}
+
+	for _, tt := range []struct {
+		name          string
+		elasticQuotas []*v1alpha1.ElasticQuota
+		nowPods       []*v1.Pod
+		nowEQUsed     []*v1alpha1.ElasticQuota
+		nextPods      []*v1.Pod
+		nextEQUsed    []*v1alpha1.ElasticQuota
```

Comment:
Maybe this reads better?

```suggestion
		existingPods       []*v1.Pod
		used     []*v1alpha1.ElasticQuota
		incomingPods      []*v1.Pod
		want    []*v1alpha1.ElasticQuota
```

Reply:
Thanks!